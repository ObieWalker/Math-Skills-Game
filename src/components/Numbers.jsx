import React from "react";
import _ from 'lodash';

const Numbers = (props) => {
  const numberClassName = (number) => {
    if (props.usedNumbers.indexOf(number) >= 0) {
      return 'used';
    }
    if (props.selectedNumbers.indexOf(number) >= 0) {
      return 'selected';
    }
  }
  
  return (
    <div className="row col-6 col-md-offset-4 card text-center">
      {Numbers.list.map((number, i) =>
      <span key={i} className={numberClassName(number)}
      onClick={()=> props.selectNumber(number)}>{number}</span>)}
    </div>
  );
};


Numbers.list = _.range(1, 10);

export default Numbers;