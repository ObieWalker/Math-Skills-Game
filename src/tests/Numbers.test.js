import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import Numbers from '../../src/components/Numbers';

let mountedForm, props, wrapper;
let NumbersList = {
  list : []
}
const selectNumber = jest.fn();
const selectedNumbers = [];
const numberClassName = jest.fn();
const usedNumbers = [];
const getComponent = () => {
  if (!mountedForm) {
    props = {
      selectNumber,
      usedNumbers,
      selectedNumbers
    };
    mountedForm = shallow(<Numbers {...props} />);
  }
  return mountedForm;
};

describe('Numbers Component', () => {
  beforeEach(() => {});

  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });
  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper.find('div').length).toBe(1);

  });
});
