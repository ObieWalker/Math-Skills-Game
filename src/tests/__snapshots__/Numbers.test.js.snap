// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Numbers Component renders component successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Numbers
    selectNumber={[MockFunction]}
    selectedNumbers={Array []}
    usedNumbers={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          onClick={[Function]}
        >
          1
        </span>,
        <span
          onClick={[Function]}
        >
          2
        </span>,
        <span
          onClick={[Function]}
        >
          3
        </span>,
        <span
          onClick={[Function]}
        >
          4
        </span>,
        <span
          onClick={[Function]}
        >
          5
        </span>,
        <span
          onClick={[Function]}
        >
          6
        </span>,
        <span
          onClick={[Function]}
        >
          7
        </span>,
        <span
          onClick={[Function]}
        >
          8
        </span>,
        <span
          onClick={[Function]}
        >
          9
        </span>,
      ],
      "className": "row col-6 col-md-offset-4 card text-center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": 1,
          "className": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": 1,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": 2,
          "className": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": 2,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": 3,
          "className": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": 3,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": 4,
          "className": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": 4,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": 5,
          "className": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": 5,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "host",
        "props": Object {
          "children": 6,
          "className": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": 6,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "host",
        "props": Object {
          "children": 7,
          "className": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": 7,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "7",
        "nodeType": "host",
        "props": Object {
          "children": 8,
          "className": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": 8,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "8",
        "nodeType": "host",
        "props": Object {
          "children": 9,
          "className": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": 9,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            onClick={[Function]}
          >
            1
          </span>,
          <span
            onClick={[Function]}
          >
            2
          </span>,
          <span
            onClick={[Function]}
          >
            3
          </span>,
          <span
            onClick={[Function]}
          >
            4
          </span>,
          <span
            onClick={[Function]}
          >
            5
          </span>,
          <span
            onClick={[Function]}
          >
            6
          </span>,
          <span
            onClick={[Function]}
          >
            7
          </span>,
          <span
            onClick={[Function]}
          >
            8
          </span>,
          <span
            onClick={[Function]}
          >
            9
          </span>,
        ],
        "className": "row col-6 col-md-offset-4 card text-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": 1,
            "className": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 1,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": 2,
            "className": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 2,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": 3,
            "className": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 3,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": 4,
            "className": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 4,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": 5,
            "className": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 5,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": 6,
            "className": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 6,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "host",
          "props": Object {
            "children": 7,
            "className": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 7,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "host",
          "props": Object {
            "children": 8,
            "className": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 8,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "8",
          "nodeType": "host",
          "props": Object {
            "children": 9,
            "className": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 9,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
